text/machine_comprehension/bert-squad/README.md

```diff
@@ -0,0 +1,84 @@
+# BERT-Squad
+
+## Use cases
+This model answers questions based on the context of the given input paragraph.   
+
+## Description
+BERT (Bidirectional Encoder Representations from Transformers) applies Transformers, a popular attention model, to language modelling. This mechanism has an encoder to read the input text and a decoder that produces a prediction for the task. This model uses the technique of masking out some of the words in the input and then condition each word bidirectionally to predict the masked words. BERT also learns to model relationships between sentences, predicts if the sentences are connected or not. 
+
+## Model
+
+ |Model        |Download  |Checksum|Download (with sample test data)| ONNX version |Opset version|Dependencies| 
+| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
+|BERT-Squad| [393,930 KB](model/bert.onnx)  |[MD5](BERT-md5.txt) |  [932 KB](model/saved_model.pb) |  1.5.2  | 8| [tokenization.py](dependencies/tokenization.py) [run_onnx_squad.py](dependencies/run_onnx_squad.py)  | 
+
+## Inference
+We used onnxruntime to preform the inference. 
+
+### Input 
+Uses WordPiece tokenisation method to split the input text into list of tokens that are available in the vocabulary (30,522 words).
+Then converts these tokens into features
+<li>input_ids: list of numerical ids for the tokenised text
+<li>input_mask: will be set to 1 for real tokens and 0 for the padding tokens
+<li>segment_ids: for our case, this will be set to the list of ones
+<li>label_ids: one-hot encoded labels for the text
+
+### Preprocessing
+Get parameters and convert input examples into features
+```python
+# preprocess input 
+predict_file = 'inputs.json'
+
+# Use read_squad_examples method from run_onnx_squad to read the input file
+eval_examples = read_squad_examples(input_file=predict_file)
+
+max_seq_length = 256
+doc_stride = 128
+max_query_length = 64
+batch_size = 1
+n_best_size = 20
+max_answer_length = 30
+
+vocab_file = os.path.join('uncased_L-12_H-768_A-12', 'vocab.txt')
+tokenizer = tokenization.FullTokenizer(vocab_file=vocab_file, do_lower_case=True)
+
+# Use convert_examples_to_features method from run_onnx_squad to get parameters from the input 
+input_ids, input_mask, segment_ids, extra_data = convert_examples_to_features(eval_examples, tokenizer,
+                                                                              max_seq_length, doc_stride, max_query_length)
+```
+
+### Output
+Given a question and a context paragraph, the model predicts a start and an end token from the paragraph that most likely answers the questions. Computes a start vector and a final transformer output for each token. 
+
+### Postprocessing
+Write the predictions (answers to the questions) in a file. 
```

Comment:
Postprocessing should be about how to turn the output of the model into something you can use. This should be independent of file or not. In fact it would be much better if we didn;t have to write out to a file.